src/Microsoft.AspNetCore.OData/Extensions/ODataServiceCollectionExtensions.cs

```diff
@@ -113,6 +113,33 @@ public static IODataBuilder AddOData(this IServiceCollection services)
             return new ODataBuilder(services);
         }
 
+        /// <summary>
+        /// Enables case-insensitive model binding.
+        /// </summary>
+        /// <param name="builder">Odata Builder.</param>
+        public static IODataBuilder EnableODataCaseInsensitiveModelBinding(this IODataBuilder builder)
+        {
+            return EnableODataCaseInsensitiveModelBinding(builder, true);
+        }
+
+        /// <summary>
+        /// Enables case-insensitive model binding.
+        /// </summary>
+        /// <param name="builder">Odata Builder.</param>
+        /// <param name="enable">Whether to enable case-insensitive model binding.</param>
+        public static IODataBuilder EnableODataCaseInsensitiveModelBinding(this IODataBuilder builder, bool enable)
+        {
+            if (builder == null)
+            {
+                throw Error.ArgumentNull("builder");
+            }
+
+            var odataOptions = builder.Services.First(s => s.ServiceType == typeof(ODataOptions) && s.ImplementationType == typeof(ODataOptions));
+            builder.Services.Remove(odataOptions);
```

Comment:
We can't simple just remove the old one, what about there's some configuration before calling this method?

So, i think we should copy construct or clone the old ODataOptions and reset the model binding property.